packages/harmony/src/components/segmented-control/SegmentedControl.tsx

```diff
@@ -15,21 +15,30 @@ import { SegmentedControlProps } from './types'
  * A hybrid somewhere between a button group, radio buttons, and tabs;
  * segmented controls are used to switch between different options or views.
  */
-export const SegmentedControl = <T extends string>(
-  props: SegmentedControlProps<T>
-) => {
-  const optionRefs = useRef(
-    props.options.map((_) => createRef<HTMLLabelElement>())
-  )
-  const [selected, setSelected] = useState(props.options[0].key)
+export const SegmentedControl = <T extends string>({
+  options,
```

Comment:
what happened to destructuring inside the component?

Reply:
ai seems to prefer this way :D, will update my cursor file